poller-lambdas/src/pollers/reuters/reutersPoller.ts

```diff
@@ -1,18 +1,299 @@
+	uri: NullishToStringOrUndefinedSchema,
+	language: NullishToStringOrUndefinedSchema,
+	type: NullishToStringOrUndefinedSchema,
+	profile: NullishToStringOrUndefinedSchema,
+	slug: NullishToStringOrUndefinedSchema,
+	usageTerms: NullishToStringOrUndefinedSchema,
+	usageTermsRole: NullishToStringOrUndefinedSchema,
+	version: NullishToStringOrUndefinedSchema,
+	credit: NullishToStringOrUndefinedSchema,
+	firstCreated: NullishToStringOrUndefinedSchema,
+	productLabel: NullishToStringOrUndefinedSchema,
+	pubStatus: NullishToStringOrUndefinedSchema,
+	urgency: z
+		.number()
+		.nullish()
+		.transform((x) => x ?? undefined),
+	usn: NullishToStringOrUndefinedSchema,
+	position: NullishToStringOrUndefinedSchema,
+	bodyXhtml: NullishToStringOrUndefinedSchema,
+	bodyXhtmlRich: NullishToStringOrUndefinedSchema,
+	subject: z.array(
+		z.object({
+			code: NullishToStringOrUndefinedSchema,
+			name: NullishToStringOrUndefinedSchema,
+			rel: NullishToStringOrUndefinedSchema,
+		}),
+	),
+});
+
+const itemResponseSchema = z.object({
+	data: z.object({
+		item: ReutersItemSchema,
+	}),
+});
+
+function itemResponseToIngestionLambdaInput(
+	item: z.infer<typeof ReutersItemSchema>,
+): IngestorInputBody {
+	const { bodyXhtmlRich, bodyXhtml } = item;
+	const bodyHtml = parse(bodyXhtmlRich ?? bodyXhtml ?? '').querySelector(
+		'body',
+	)?.innerHTML;
+	return {
+		originalContentText: item.bodyXhtmlRich,
+		uri: item.uri,
+		'source-feed': 'Reuters-Newswires',
+		usn: item.usn,
+		version: item.version,
+		type: item.type,
+		firstVersion: item.firstCreated,
+		versionCreated: item.versionCreated,
+		dateTimeSent: item.versionCreated,
+		originalUrn: item.versionedGuid,
+		slug: item.slug,
+		headline: item.headLine,
+		byline: item.byLine,
+		priority: item.urgency?.toString() ?? '',
+		subjects: {
+			code: item.subject
+				.map((subject) => subject.code)
+				.filter((_): _ is string => _ !== undefined),
+		},
+		mediaCatCodes: '',
+		keywords: [],
+		organisation: { symbols: [] },
+		tabVtxt: 'X',
+		status: item.pubStatus,
+		usage: item.usageTerms,
+		ednote: '',
+		abstract: item.fragment,
+		body_text: bodyHtml,
+		copyrightNotice: item.copyrightNotice,
+		language: item.language,
+	};
+}
+
+const SecretValueSchema = z.object({
+	CLIENT_ID: z.string(),
+	CLIENT_SECRET: z.string(),
+	ACCESS_TOKEN: z.string().optional(),
+});
 
 export const reutersPoller = (async (
-	_secret: SecretValue,
-	_input: PollerInput,
+	secret: SecretValue,
+	input: PollerInput,
 ) => {
-	console.log('Reuters poller running');
-	await new Promise((resolve) => setTimeout(resolve, 1000));
+	const parsedSecret = SecretValueSchema.safeParse(JSON.parse(secret));
+	if (!parsedSecret.success) {
+		throw new Error('Failed to parse secret value for Reuters poller');
+	}
+	const { CLIENT_ID, CLIENT_SECRET, ACCESS_TOKEN } = parsedSecret.data;
+
+	let accessToken = ACCESS_TOKEN ?? (await auth(CLIENT_ID, CLIENT_SECRET));
+
+	async function fetchWithReauth(query: string) {
```

Comment:
this function is defined within the scope of the main poller function so that it can read and mutate `accessToken`, I went back and forth on whether to handle this in a more 'pure' way, but as mentioned in the PR description this felt like a good balance in context. Open to other perspectives though!